lib/security/JweToken.ts

```diff
@@ -74,6 +118,84 @@ export default class JweToken extends JoseToken {
     return Buffer.from(jweString);
   }
 
+  /**
+   * Encrypts the given string in JWE JSON serialized format using the given key in JWK JSON object format.
+   * Content encryption algorithm is hardcoded to 'A128GCM'.
+   *
+   * @returns Encrypted Buffer in JWE compact serialized format.
+   */
+  public async flatJsonEncrypt (jwk: PublicKey,
+    options?: {
+      unprotected?: {[key: string]: string},
+      protected?: {[key: string]: string},
+      aad?: string | Buffer
+    }
+    /* encryptionType?: string */): Promise<{
+      protected?: string,
+      unprotected?: {[key: string]: string},
+      encrypted_key: string,
+      iv: string,
+      ciphertext: string,
+      tag: string,
+      aad?: string
+    }> {
+    /// TODO: extend to include encryptionType to determine symmetric key encryption using register
```

Comment:
Consider creating an issue and reference it for every TODO comment for easy tracking.